packages/grid/_modules_/grid/components/columnHeaders/GridColumnHeaderItem.tsx

```diff
@@ -90,16 +94,33 @@ export const GridColumnHeaderItem = ({
     [apiRef, column.field],
   );
 
+  const publishClick = React.useCallback(
+    (eventName: string) => (event: React.MouseEvent) => {
+      apiRef!.current.setCellFocus({
+        colIndex,
+        rowIndex: null,
+      });
+
+      apiRef!.current.publishEvent(
+        eventName,
+        apiRef!.current.getColumnHeaderParams(column.field),
+        event,
+      );
+    },
+    [apiRef, column.field, colIndex],
+  );
+
   const mouseEventsHandlers = React.useMemo(
     () => ({
-      onClick: publish(GRID_COLUMN_HEADER_CLICK),
+      onClick: publishClick(GRID_COLUMN_HEADER_CLICK),
       onDoubleClick: publish(GRID_COLUMN_HEADER_DOUBLE_CLICK),
       onMouseOver: publish(GRID_COLUMN_HEADER_OVER),
       onMouseOut: publish(GRID_COLUMN_HEADER_OUT),
       onMouseEnter: publish(GRID_COLUMN_HEADER_ENTER),
       onMouseLeave: publish(GRID_COLUMN_HEADER_LEAVE),
+      onKeyDown: publish(GRID_CELL_KEYDOWN),
```

Comment:
it's not a cell

Reply:
I think we should treat it as one. The event can be renamed but I think we should draw a line between a column header and a cell. They are the same but have different purposes. For example, if one day we need to make the column headers editable that would mean that we would probably need to implement the edit logic again. We should discuss it.

Reply:
Here you need to refactor to `GRID_COLUMN_HEADER_KEYDOWN`

Reply:
Then add a `handleColumnHeaderKeyDown` method in `useGridKeyboard`
then check that if it's arrow down key, `navigateCell`
And if you're on row 0 and arrow up, `navigateHeader`

Reply:
I'm ok with the separate handler and events in the `useGridKeyboard` but why do we need a separate one for the `useGridKeyboardNavigation`. Navigation cells is a generic action that we should be able to select each column header regardless if it has actions or not, at least IMO. It looks that a separate handler will have exactly the same check as in the `navigateCells` and still use `setCellFocus`.

Reply:
updated